### PR TITLE
Feat target provider type

### PR DESCRIPTION
**Pull request by fanatic75**
## What does this PR do?

Adds provider type in targets and makes providerId as optional unless type is push

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Screenshots may also be helpful.)

## Related PRs and Issues

- (Related PR or issue)

## Checklist

- [ ] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [ ] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

File: app/controllers/api/account.php [hunk 1]
```diff
@@ -2019,6 +2049,27 @@
...
+
+        /**
+        * @var Document $oldTarget
+        */
+        $oldTarget = $user->find('identifier', $oldEmail, 'targets');
+
+        if ($oldTarget !== false && !$oldTarget->isEmpty()) {
+            try {
+                $dbForProject->updateDocument('targets', $oldTarget->getId(), $oldTarget->setAttribute('identifier', $email));
+            } catch (Duplicate) {
```
Comment (abnegate) [inline app/controllers/api/account.php:2068]: Do we need the try/catch? I think duplicate should only be thrown on create

File: app/controllers/api/messaging.php [hunk 2]
```diff
@@ -806,14 +806,21 @@
...
+        if ($enabled === true || $enabled === false) {
+            if ($provider->getAttribute('internal') === true && $enabled === false) {
+                throw new Exception(Exception::PROVIDER_INTERNAL_UPDATE_DISABLED);
+            }
+            if ($provider->getAttribute('internal') === true && $enabled === false) {
+                throw new Exception(Exception::PROVIDER_INTERNAL_UPDATE_DISABLED);
+            }
+
+            $provider->setAttribute('enabled', $enabled);
+        }
```
Comment (abnegate) [inline app/controllers/api/messaging.php:822]: Let's remove the outer `if` here

File: app/controllers/api/messaging.php [hunk 3]
```diff
@@ -806,14 +806,21 @@
...
             $provider->setAttribute('internal', $internal);
         }
 
+        if ($enabled === true || $enabled === false) {
+            if ($provider->getAttribute('internal') === true && $enabled === false) {
+                throw new Exception(Exception::PROVIDER_INTERNAL_UPDATE_DISABLED);
+            }
+            if ($provider->getAttribute('internal') === true && $enabled === false) {
+                throw new Exception(Exception::PROVIDER_INTERNAL_UPDATE_DISABLED);
+            }
```
Comment (abnegate) [inline app/controllers/api/messaging.php:819]: Looks like a duplicate here

Comment (fanatic75) [reply app/controllers/api/messaging.php]: fixed this

File: app/controllers/api/account.php (hunk 1 above)
Comment (fanatic75) [reply app/controllers/api/account.php]: removed this

File: app/controllers/api/messaging.php (hunk 2 above)
Comment (fanatic75) [reply app/controllers/api/messaging.php]: need to keep it

**Review by abnegate — APPROVED** [review]

**Pull request merged**